addons/sms/tests/test_sms_template.py

```diff
@@ -106,3 +106,28 @@ def test_sms_template_rendering_unrestricted(self):
 
         body = sms_composer._prepare_body_values(self.partner)[self.partner.id]
         self.assertIn(self.partner.name, body, 'Template Editor should be able to write new Jinja code')
+
+@tagged("post_install")
+class TestSMSComposer(TransactionCase):
```

Comment:
We have addons specifically for mail/sms tests. This one should go in ``test_mail_sms``, in ``test_sms_composer.py``

Reply:
Hi, I tried this in test_sms_composer.py but I get an error, I don't know why
```python
    def test_composer_sending_on_sale_order(self):
        partner = self.env['res.partner'].create({'name': 'Test Partner'})
        sale_order = self.env['sale.order'].create({
            'partner_id': partner.id,
        })

        sms_composer = self.env['sms.composer'].create({
            'body': self._test_body,
            'composition_mode': 'comment',
            'mass_force_send': False,
            'mass_keep_log': True,
            'number_field_name': False,
            'numbers': False,
            'recipient_single_number_itf': self.random_numbers_san[0],
            'res_id': sale_order.id,
            'res_model': 'sale.order'
        })

        with self.mockSMSGateway():
            sms_composer._action_send_sms()

        self.assertSMSNotification([{'partner': partner, 'number': self.random_numbers_san[0]}], self._test_body)

``` 
![Screenshot from 2023-01-09 15-53-10](https://user-images.githubusercontent.com/97248555/211336958-85bc1f26-552a-4535-931a-350c8af38d27.png)

Can you help me ? :)

Reply:
You should not use 'sale.order' as a test record, as it is not a dependency of this module. Use one of the test models available, tde pointed out ``mail.test.sms.partner`` was great for this.

You don't need to use a partner either, you can just do 
``test_record = self.env['mail.test.sms.partner'].create({'name': 'Test'})`` (update the composer too)
and 
``self.assertSMSNotification([{'number': self.random_numbers_san[0]}], self._test_body)``